src/audio_core/hle/adts.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#ifndef ADTS_ADT
+#define ADTS_ADT
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+struct ADTSData {
+    bool MPEG2;
+    uint8_t profile;
+    uint8_t channels;
+    uint8_t channel_idx;
+    uint8_t framecount;
+    uint8_t samplerate_idx;
+    uint32_t length;
+    uint32_t samplerate;
+};
+
+typedef struct ADTSData ADTSData;
+
+#ifdef __cplusplus
+extern "C" {
+#endif // __cplusplus
+uint32_t parse_adts(char* buffer, struct ADTSData* out);
```

Comment:
I think this can be just a (explicit) constructor of ADTSData, so we can avoid all the extern C stuff.

Reply:
this isn't resolved